### PR TITLE
Pull Request for Issue1832: Adding the QE65000 to the SGM

### PR DESCRIPTION
**Pull request by iainworkman**
- Added the QE65000 detector to the SGM Beamline.
- Added the detector view to the SGM App Controller.
- Fixed a reversed logic bug in CLSQE65000Detector

### CODE REVIEW

File: source/beamline/CLS/CLSQE65000Detector.cpp [hunk 1]
```diff
@@ -158,12 +158,18 @@ bool CLSQE65000Detector::initializeImplementation(){
 }
 
 bool CLSQE65000Detector::acquireImplementation(AMDetectorDefinitions::ReadMode readMode){
-	if(!isConnected() || readMode != AMDetectorDefinitions::SingleRead)
+	if(!isConnected() || readMode != AMDetectorDefinitions::SingleRead) {
+
+		if(readMode != AMDetectorDefinitions::SingleRead) {
```
Comment (dretrex) [inline source/beamline/CLS/CLSQE65000Detector.cpp]: Is that return value supposed to be inside those curly braces?

File: source/beamline/CLS/CLSQE65000Detector.cpp [hunk 2]
```diff
@@ -158,12 +158,18 @@ bool CLSQE65000Detector::initializeImplementation(){
...
+		}
 		return false;
+	}
 
 	AMControl::FailureExplanation failureExplanation = startAcquisitionControl_->move(1);
-	if(failureExplanation != AMControl::NoFailure)
+	if(failureExplanation == AMControl::NoFailure) {
```
Comment (dretrex) [inline source/beamline/CLS/CLSQE65000Detector.cpp]: So was this broken the whole time?  This seems like a pretty big deal to switch from `!=` to `==`.

**Comment by dretrex** [comment]
Couple of comments.
Time Estimate: 2 minutes

File: source/beamline/CLS/CLSQE65000Detector.cpp (hunk 1 above)
Comment (iainworkman) [inline source/beamline/CLS/CLSQE65000Detector.cpp]: Nope, I added that conditional in order to put a debug in there to show the reason why this call was returning false. When I cleaned up the code ready for submission I got rid of all the debugs, but forgot to remove this conditional block.

File: source/beamline/CLS/CLSQE65000Detector.cpp (hunk 2 above)
Comment (iainworkman) [inline source/beamline/CLS/CLSQE65000Detector.cpp]: Until 3e31c42c2fee83d181a425dc770faf9f9a9a7db3 the AMDetectorTriggerAction didn't pay attention to the return result of the call to the detector's acquire() function. The action succeeded either way. So while this function was returning the wrong value nothing was paying attention to it.

Comment (dretrex) [inline source/beamline/CLS/CLSQE65000Detector.cpp]: Ah.  Well, that's a bit troubling.  Good catch then.

**Pull request merged**